### PR TITLE
Fix pan limit

### PR DESCRIPTION
**Pull request by nikordaris**
Fixes issue #30 by calculating the max Math.abs() of cx and cy of the node positions to use as the height and width of the ZoomableSVGGroup instead of the canvas height and width. This ensures that panning will include all of the data on the canvas

### CODE REVIEW

**Comment by fastfrwrd** [comment]
@nikordaris please sign the CLA (see the checks) and update your reduce logic, and I think we're good to go.

**Comment by nikordaris** [comment]
@fastfrwrd I signed the CLA. For some reason i keep having to click the link to recheck.

File: src/components/ForceGraph.js [hunk 1]
```diff
@@ -251,7 +258,10 @@ export default class ForceGraph extends PureRenderComponent {
     const linkElements = [];
     const zoomableChildren = [];
     const staticChildren = [];
-
+    const maxPanWidth = reduce(nodePositions, (maxWidth, { cx }) =>
+      (maxWidth > Math.abs(cx) ? maxWidth : Math.abs(cx)), 0);
+    const maxPanHeight = reduce(nodePositions, (maxHeight, { cy }) =>
+      (maxHeight > Math.abs(cy) ? maxHeight : Math.abs(cy)), 0);
```
Comment (fastfrwrd) [inline src/components/ForceGraph.js]: nodePositions.reduce(....) for both of these to get rid of lodash dep

Comment (fastfrwrd) [inline src/components/ForceGraph.js]: I can make this reduce change to the PR, I believe.

**Comment by fastfrwrd** [comment]
tried to make that change but realized that you were right to pull that in, it's totally an object and I'd forgotten. I'd prefer it if we did the following, but I'll land this as-is and add that separately.

```
diff --git a/package.json b/package.json
index fa08e1e..d40ef97 100644
--- a/package.json
+++ b/package.json
@@ -68,7 +68,7 @@
   "dependencies": {
     "d3-force": "^1.0.2",
     "global": "^4.3.0",
-    "lodash": "^4.17.4",
+    "lodash.reduce": "^4.6.0",
     "react-addons-shallow-compare": "0.14.x - 15.x"
   },
   "scripts": {
diff --git a/src/components/ForceGraph.js b/src/components/ForceGraph.js
index 388dbd9..9caad41 100644
--- a/src/components/ForceGraph.js
+++ b/src/components/ForceGraph.js
@@ -19,7 +19,7 @@
 // THE SOFTWARE.
 
 import React, { PropTypes, Children, cloneElement } from 'react';
-import { reduce } from 'lodash';
+import reduce from 'lodash.reduce';
 
 import './ForceGraph.css';
 import PureRenderComponent from './PureRenderComponent';

```

**Pull request merged**